realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -204,6 +204,14 @@ public RealmObjectSchema removeField(String fieldName) {
         if (table.getPrimaryKey() == columnIndex) {
             table.setPrimaryKey(null);
         }
+        // When PK does not exist (-2) this cannot happen. Meanwhile a removal target has a smaller
```

Comment:
Can you expand this comment because I don't understand it?
Primary keys should be saved by their field name, so removing a field should not impact the primary key stored in the pk table as far as I know?